### ISSUE TITLE
Setting up PyTorch plugin "upfirdn2d_plugin"... Failed!

### ISSUE THREAD

**Issue by ztisdashen** [NVlabs/stylegan2-ada-pytorch#215]
~/.cache/torch_extensions/upfirdn2d_plugin/upfirdn2d_plugin.so: cannot open shared object file: No such file or directory

  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n\n' + traceback.format_exc())

linux
cuda:11.1
gpu:3090
torch:1.9

**Comment by jexz11** [comment]
> 
Hello, have you solved this problem yet? I had the exact same problem

**Comment by MationPlays** [comment]
> you need to create the .so file

what is an .so file and where to find it?

**Comment by yllgl** [comment]
> > you need to create the .so file
> 
> what is an .so file and where to find it?

I have same problems. Help !!!!!!!!!

**Comment by Xjmengnieer** [comment]
This is your environmental problem. Check the virtual environment carefully

**Comment by MationPlays** [comment]
> > > you need to create the .so file
> > 
> > 
> > what is an .so file and where to find it?
> 
> I have same problems. Help !!!!!!!!!

I know this is not really a fix, but I never had this problem training the model in a ubuntu machine or on google colab, only on windows

**Comment by MationPlays** [comment]
> > > > you need to create the .so file
> > > 
> > > 
> > > what is an .so file and where to find it?
> > 
> > 
> > I have same problems. Help !!!!!!!!!
> 
> This is your environmental problem. Check the virtual environment carefully

I set up the env with this on windows in an anaconda enviroment:
conda install pytorch==1.7.1 torchvision==0.11.2 torchaudio==0.10.1 cudatoolkit=11.3 -c pytorch -c conda-forge

pip install torch===1.7.1+cu110 torchvision===0.8.2+cu110 torchaudio===0.7.2 -f https://download.pytorch.org/whl/torch_stable.html

pip install click requests tqdm pyspng psutil scipy ninja imageio-ffmpeg==0.4.3

I do the same on a Ubuntu machine and it works, just confusing

**Comment by MationPlays** [comment]
> > > > you need to create the .so file
> > > 
> > > 
> > > what is an .so file and where to find it?
> > 
> > 
> > I have same problems. Help !!!!!!!!!
> 
> This is your environmental problem. Check the virtual environment carefully

I set up the env on windows like this:
conda install pytorch==1.7.1 torchvision==0.11.2 torchaudio==0.10.1 cudatoolkit=11.3 -c pytorch -c conda-forge

pip install torch===1.7.1+cu110 torchvision===0.8.2+cu110 torchaudio===0.7.2 -f https://download.pytorch.org/whl/torch_stable.html

pip install click requests tqdm pyspng psutil scipy ninja imageio-ffmpeg==0.4.3

Doing the same on an ubuntu machine and it works, but on windows the Setting up PyTorch plugin "upfirdn2d_plugin"... failed Error

**Comment by Xjmengnieer** [comment]
> 
This problem will be very troublesome to solve. Changing a server and creating a new environment may solve everything

**Comment by MationPlays** [comment]
> > 
> 
> This problem will be very troublesome to solve. Changing a server and creating a new environment may solve everything

Yeah I also did this with PATH you have to do when training on windows: "C:\Program Files (x86)\Microsoft Visual Studio\<VERSION>\Community\VC\Auxiliary\Build\vcvars64.bat".

Creating a new env won't solve the problem I tried it multiple times. At this I do not know how to solve it but I just switched training with a ubuntu VM which works fine

**Comment by MationPlays** [comment]
> > > you need to create the .so file
> > 
> > 
> > what is an .so file and where to find it?
> 
> I have same problems. Help !!!!!!!!!

This is also a fixed I tried, did not work for me but maybe you:

'One known problem with PyTorch extension builds is that breaking the build may sometimes leave a lock file under $HOME/.cache/torch_extensions (Linux). You can try to delete this whole directory and try again.

Please refer to PyTorch documentation to find out where the torch_extensions directory resides if you can't find it. I'm not sure where it gets put on Windows.'

**Comment by seawardhai** [comment]
> 

您好，我在Linux上遇到了您在window上遇到的问题。

**Comment by gregschuit** [comment]
Any updates on this?

**Comment by MationPlays** [comment]
> 

My problem was I was logged into the wrong user in the terminal, I was logged into root user where the pytorch extensions weren‘t installed

**Comment by srd2018** [comment]
> 

我也是，请问你解决了吗？

**Comment by srd2018** [comment]
> > 
> 
> 您好，我在Linux上遇到了您在window上遇到的问题。

我也是，请问你解决了吗？

**Comment by Zhuhh0311** [comment]
You can refer to this issue  https://github.com/zhou13/neurvps/issues/1

**Comment by LexTran** [comment]
I have solved this problem finally. For those who still have encountered this problem, I have wrote a blog for solving this issue. Here is the [address](https://lextran.github.io/stylegan/).

**Comment by 3er1inWa11** [comment]
运行时提示ImportError: DLL load failed while importing upfirdn2d_plugin: 找不到指定的模块。

  warnings.warn('Failed to build CUDA kernels for upfirdn2d. Falling back to slow reference implementation. Details:\n\n' + traceback.format_exc())，即使我几天前已经配置好环境并成功运行，但是不知道为什么过了两天就出现了这一问题，请问有人可以解决吗

**Comment by bella-wong** [comment]
pip install ninja

problem solved.
